RevenueCat/Scripts/Purchases.cs

```diff
@@ -140,14 +131,20 @@ private void Configure(string newUserId)
                  || IsAndroidEmulator())
             apiKey = useAmazon ? revenueCatAPIKeyAmazon : revenueCatAPIKeyGoogle;
 
+        if (purchasesAreCompletedBy == PurchasesAreCompletedBy.MyApp && storeKitVersion == StoreKitVersion.Default)
+        {
+            Debug.Log("You must set a StoreKit version if you are setting PurchasesAreCompletedBy to MyApp. For Android, it doesn't matter which");
```

Comment:
A couple things to note here, when configuring through the Editor:
- There is no breaking change even after removing the old observer mode. So if someone was setting observer mode in the editor before, there won't be any compilation issue when they migrate to the new version.
- There is also no compile-time check that makes it so you can't set "PurchasesAreCompletedBy.MyApp" without setting storeKitVersion as well.
For these 2 reasons, we're adding this runtime check which would make it so we don't configure the SDK in this situation and log this message.